common.py

```diff
@@ -928,37 +1027,37 @@ def split_bdt(sel):
             print(f"Invalid number {parts[1]} following 'bdt='.")
     else:
         raise InvalidSelectionException(sel=selection)
+    return parts[1]
 
-def apply_bdtbased(cols,wp,lumi):
+def calc_bdt_scores(X, model_file=bdt_model_file):
     import xgboost as xgb
 
+    # Load the model and get the predictions
+    xgb_model = xgb.XGBClassifier()
+    xgb_model.load_model(model_file)
+    with time_and_log(f'Calculating xgboost scores for {bdt_model_file}...'):
+        score = xgb_model.predict_proba(X)[:,1]
+    return score
+
+def apply_bdtbased(cols,wp,lumi):
     cols = apply_rt_signalregion(cols)
 
-    # Grab the input features and weights
+    # Grab the weights and scores
     X = []
+    score = {}
     weight = []
 
     # Get the features for the bkg samples
-    X = cols.to_numpy(bdt_features)
-    # Load the model and get the predictions
-    xgb_model = xgb.XGBClassifier()
-    xgb_model.load_model(bdt_model_file)
-    with time_and_log(f'Calculating xgboost scores for {bdt_model_file}...'):
-        score = xgb_model.predict_proba(X)[:,1]
-    weight = (cols.xs / cols.cutflow['raw']) * lumi * cols.arrays['puweight']
-    print('weight length: ', len(weight), ' weight: ', weight)
-
-    # Obtain the efficiencies for the desired BDT working point
-    # bdt_cut is the user input bdt_cut
-    bdt_Hist=np.histogram(score[score>bdt_cut],weights=weight[score>bdt_cut]*len(score))
-    bdt_Hist_nom=np.histogram(score[score>0.0],weights=weight[score>0.0]*len(score))
-    eff = sum(bdt_Hist[0])/sum(bdt_Hist_nom[0])
+    X = cols.to_numpy(training_features)
```

Comment:
this looks like a typo: it was `X = cols.to_numpy(bdt_features)`, and `training_features` is not a defined variable in this function.